plugins/catalog-backend/src/service/createRouter.ts

```diff
@@ -228,6 +231,28 @@ export async function createRouter(
     });
   }
 
+  if (orchestrator) {
+    router.post('/validate-entity', async (req, res) => {
+      const bodySchema = z.object({
+        entity: z.unknown(),
+        location: z.string(),
+      });
+
+      const body = await validateRequestBody(req, bodySchema);
+
+      const entity = setRequiredEntityLocationAnnotations(
+        body.entity as Entity,
+        body.location,
+      );
```

Comment:
Tbh I think it's better to just inline this bit and use a manual object spread instead. Less jumping around and pretty clear what it does :grin:

```suggestion
    const entity = {
      ...body.entity,
      metadata: {
        ...body.entity.metadata,
        annotations: {
          [ANNOTATION_LOCATION]: body.location,
          [ANNOTATION_ORIGIN_LOCATION]: body.location,
          ...body.entity.metadata.annotations,
        }
      }
    }
```

Btw, I believe we should be setting `ANNOTATION_LOCATION` and `ANNOTATION_ORIGIN_LOCATION` instead.